src/v/security/gssapi_authenticator.cc

```diff
@@ -419,4 +421,51 @@ void gssapi_authenticator::fail_impl(
     _state = state::failed;
 }
 
+acl_principal
+gssapi_authenticator::get_principal_from_name(std::string_view source_name) {
+    krb5::krb5_context krb5_ctx;
+    int krb5_rv = ::krb5_init_context(&krb5_ctx);
+    if (krb5_rv != 0) {
+        vlog(
+          seclog.error,
+          "Failed to initialize KRB5 instance for source_name mapping: {}",
+          krb5_rv);
+        return {};
+    }
+
+    const krb5::krb5_context_view krb5_ctx_view(&krb5_ctx);
+
+    krb5::krb5_default_realm default_realm(krb5_ctx_view);
+
+    krb5_rv = ::krb5_get_default_realm(
```

Comment:
and this one seems to at least console printing and file i/o, at least if we are compiled with USE_DLOPEN. wdyt @BenPope ?

Reply:
also :alien: